code/framework/engine/src/main/java/io/cattle/platform/engine/eventing/impl/ProcessEventListenerImpl.java

```diff
@@ -81,6 +91,25 @@ public void processExecute(Event event) {
         }
     }
 
+    private boolean shouldWaitLonger(long processId) {
+        ProcessRecord record = processRecordDao.getRecord(processId);
+        if (record == null || record.getEndTime() != null) {
+            // is there any additional processing that happens or can we
+            // even short circuit it here and be done with this event?
+            // for now, assume additional processing occurs to get rid
+            // of event
+            return false;
+        }
+        int numPreviousAttempts = processRecordDao.getNumPreviousExecutions(processId);
+        if (numPreviousAttempts < 2) {
+            return false;
+        }
+
+        long now = System.currentTimeMillis();
+        long lastExecution = processRecordDao.getLastExecutionTimestamp(processId);
+        return now < lastExecution + Math.pow(2,  numPreviousAttempts) * 15000;
```

Comment:
We can tweak this too if we want.  I guess I should've extract this into an env variable.